content/articles/2018-11-tibble-2.0.0-pre-announce.Rmd

```diff
@@ -0,0 +1,222 @@
+
+In tibble v1.4.2, `as_tibble()` has a `validate` argument, but its default behaviour value was inconsistent across different methods and there was no equivalent argument for `tibble()`. The `validate` argument is now soft-deprecated and its use will trigger a message, once per session. The `validate` argument will eventually be removed, but for now it can be used jointly with the new `.name_repair` argument (without even triggering a message). This is possible, because fortunately the `.name_repair` argument to `as_tibble()` is ignored in tibble v1.4.2.
+
+Here's what `validate` does in the development version of tibble for tibbles, data frames, and matrices, along with suggested alternatives.
+
+### Tibbles and data frames
+
+The default was `validate = FALSE` for tibbles and `validate = TRUE` for data frames.
+Code that worked before for tibbles can now throw unexpected errors if the resulting tibble has problematic names.
+Use the default instead of `validate = TRUE`, and `.name_repair = "minimal"` in addition to `validate = FALSE`.
+If your code targets tibble >= v2.0.0, you can remove the `validate` argument.
+
+```{r}
+df <- new_tibble(list(a = 5, a = 6), nrow = 1)
+
+# errors, as it should, because names are duplicated ... but also messages
+as_tibble(df, validate = TRUE)
+
+# errors due to default .name_repair = "check_unique"
+# (but no error in tibble v1.4.2)
+as_tibble(df)
+
+# ensures that the validate = TRUE default is used for tibble < 2.0.0
+as_tibble(as.data.frame(df))
+
+
+# no error ... but still messages
+as_tibble(df, validate = FALSE)
+
+# no error, quietly
+as_tibble(df, .name_repair = "minimal")
+
+# no error, quietly, compatible with tibble < 2.0.0
+as_tibble(df, validate = FALSE, .name_repair = "minimal")
+```
+
+
+### Matrices and other objects
+
+The `validate` argument was silently ignored in v1.4.2, now triggers a message.
+Remove the `validate` argument, or add a consistent `.name_repair` argument.
+If you need anything other than `"minimal"` or `"check_unique"` and need to keep the `validate` argument, rename the columns beforehand.
+
+```{r}
+m <- cov(iris[-5])
+# Assign colnames() if necessary
+as_tibble(m)
+as_tibble(m, validate = TRUE, .name_repair = "check_unique")
+```
+
+
+## Deprecation of `tidy_names()` and `set_tidy_names()`
+
+The existing `tidy_names()` and `set_tidy_names()` functions are soft-deprecated, but remain available, unchanged. In the future, they could go away or take on a new meaning, i.e. implement a different algorithm for name repair. New code should use `.name_repair` instead.
+
+```{r}
+df <- new_tibble(list(a = 5, a = 6), nrow = 1)
+
+# these functions are soft-deprecated
+tidy_names(names(df))
+set_tidy_names(df)
+
+# achieve same via `.name_repair`
+as_tibble(df, .name_repair = "universal")
+```
+
+tibble's name repair strategies are currently only exposed in `tibble()` and `as_tibble()`, not (yet?) as utility functions that operate on a vector of names.
```

Comment:
Take out the "(yet?)" for the final post?

Reply:
The "yet" refers to future releases of _tibble_, I don't mind keeping it. What do you think, @jennybc?